### PR TITLE
KeePassXC-devel: fix compilation on older systems (update 1)

### PR DESCRIPTION
**Pull request by tenzap**
* remove patch that was merged upstream + update to latest devel commit
* Try to fix compilation issues on osx 10.7-10.11

This is called "update 1" in case other failures are revealed (once this PR is merged) and need fixes in the near future. (no other change is planed on this PR until merge)

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G65
Xcode 9.4.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by pmetzger** [comment]
Do you know why the CI failed?

File: security/KeePassXC/Portfile [hunk 1]
```diff
@@ -50,16 +50,15 @@ if {${subport} eq ${name}} {
                         patch-old-mac.diff
 } else {
     # devel subport
-    github.setup        keepassxreboot keepassxc a3dc977e58470644f2acca77905285d44b22f2b8
+    github.setup        keepassxreboot keepassxc d3b28f86515df73194d1102253b739b51b1909f5
     set githash         [string range ${github.version} 0 6]
-    version             20211122+git${githash}
-    revision            1
+    version             20211123+git${githash}
```
Comment (pmetzger) [inline security/KeePassXC/Portfile:55]: You might want to keep a "revision 0" there.

**Comment by tenzap** [comment]
> Do you know why the CI failed?

Missing Dependency. No Reason for that. Probably a temporary problem.

File: security/KeePassXC/Portfile (hunk 1 above)
Comment (tenzap) [reply security/KeePassXC/Portfile:55]: Initially there was no revision at all and it seems the policy on this varies in time. Some prefer to have it removed when version changes others prefer to keep it with 0. So how to know what is expected in which situation?

Btw it is likely there will be a future revision soon anyway either with additional fixes (depending on whether other build issues will reveal), or once upstream merged the pending pr.

Shall I recommit with that single change or can it wait for the next PR ?

Comment (pmetzger) [reply security/KeePassXC/Portfile:55]: The policy changed. Originally we did not leave "revision 0" in Portfiles but now we do. This is a consistent policy across the project.

It is fine to ask for multiple small pulls. In fact, that makes review easier. Regardless, do you wish for me to merge this now?

Comment (tenzap) [reply security/KeePassXC/Portfile:55]: Ok thanks for clarification.
Yes please, merge this.

Comment (pmetzger) [reply security/KeePassXC/Portfile:55]: Could you put the "revision 0" in first? As soon as you push that I'll merge.

Comment (tenzap) [reply security/KeePassXC/Portfile:55]: Done. Thanks :)

**Comment by mascguy** [comment]
I'll test these changes on a few of my macOS VMs, 10.7 and 10.11 at a minimum. More to follow

**Comment by mascguy** [comment]
> Great! That will be sufficient to begin with as one problem is in 10.10-10.11 (with AppleClang 7 & 8) & the other in 10.7-10.9 (with clang-mp-9)

Tested on 10.7, 10.9, and 10.11, and builds succeeded in all three. Great work!

**Pull request merged**